fastlane/Fastfile

```diff
@@ -63,50 +67,106 @@ platform :ios do
-in xcode build configuration and just reflected in `Info.plist` when building."
+  desc "- Discards the changes in `Info.plist`. Given this file is used for every configuration, these values are stored \
+in xcode user defined build configuration and just reflected in `Info.plist` during building."
   desc "- Uploads the application to `TestFlight` using `pilot`."
   desc ""
-  desc "NOTE: It's important to note that in case the the version number needs to be updated, it has to be manually done, \
-editing the `'CFBundleShortVersionString'` key from `Info.plist`."
-  lane :release_testflight do
+  desc "Receives a parameter `bump_type` representing the type of deploy. It can be any of %s" % release_bump_types.to_s
+  desc "Build is initialized in %s, and version in %s, so first deploy must be %s" % [first_build, first_version, release_bump_types.last]
+  lane :release_testflight do |options|
 
-    desc "Read the build number from project build settings."
-    current_build_number = read_project_property(
-      build_configuration: build_configurations[:testflight],
-      build_setting: 'CURRENT_PROJECT_VERSION'
+    desc "Back up version number from `Info.plist`."
+    info_plist_backup_version_number = get_info_plist_value(
+      path: project_plist_path % ENV['PROJECT_NAME'],
+      key: 'CFBundleShortVersionString'
     )
 
-    next_build_number = String(current_build_number.to_i + 1)
+    desc "Back up build number from `Info.plist`."
+    info_plist_backup_build_number = get_info_plist_value(
+      path: project_plist_path % ENV['PROJECT_NAME'],
+      key: 'CFBundleVersion'
+    )
 
-    desc "Update the build number in project build settings."
-    update_project_property(
+    desc "Read current version number from project configuration."
+    current_version_version = read_project_property(
       build_configuration: build_configurations[:testflight],
-      build_setting: 'CURRENT_PROJECT_VERSION',
-      build_setting_value: next_build_number
+      build_setting: 'CURRENT_VERSION_NUMBER'
     )
 
-    desc "Back up the build number in `Info.plist`."
-    info_plist_backup_build_number = get_info_plist_value(
+    desc "Read current build number from project configuration."
+    current_build_version = read_project_property(
+      build_configuration: build_configurations[:testflight],
+      build_setting: 'CURRENT_BUILD_NUMBER'
+    )
+
+    desc "Get the type of release desired"
+    bump_type = options[:bump_type]
+    while not release_bump_types.include? bump_type do
+      bump_type = UI.input "Choose the `bump_type` representing the type of deploy. It can be any of %s" % release_bump_types.to_s
+    end
+
+    if current_version_version == first_version && bump_type != release_bump_types.last
+      UI.user_error! "This seems to be your first deploy. It must be done as %s." % release_bump_types.last
```

Comment:
Right. Maybe maybe first ask him if he wants to make it a major release and if he answers no, then end the program. I don't know, if you think it's useful.